src/3D/Camera.cpp

```diff
@@ -9,28 +9,136 @@
 
 #include "Camera.h"
 #include "3D/LandIsland.h"
+#include "ECS/Components/Transform.h"
 #include "ECS/Registry.h"
+#include "ECS/Systems/DynamicsSystem.h"
 #include "Game.h"
 
 #include <glm/gtx/euler_angles.hpp>
+#include <glm/gtx/intersect.hpp>
+#include <glm/gtx/norm.hpp>
+#include <glm/gtx/spline.hpp>
+#include <optional>
 
 using namespace openblack;
 
-glm::mat4 Camera::getRotationMatrix() const
+Camera::Camera(glm::vec3 position, glm::vec3 rotation)
+    : _position(position)
+    , _rotation(glm::radians(rotation))
+    , _dv(0.0f, 0.0f, 0.0f)
+    , _dwv(0.0f, 0.0f, 0.0f)
+    , _dsv(0.0f, 0.0f, 0.0f)
+    , _ddv(0.0f, 0.0f, 0.0f)
+    , _duv(0.0f, 0.0f, 0.0f)
+    , _drv(0.0f, 0.0f, 0.0f)
+    , _projectionMatrix(1.0f)
+    , _velocity(0.0f, 0.0f, 0.0f)
+    , _hVelocity(0.0f, 0.0f, 0.0f)
+    , _rotVelocity(0.0f, 0.0f, 0.0f)
+    , _accelFactor(0.001f)
+    , _movementSpeed(4.0f)
+    , _maxMovementSpeed(0.005f)
+    , _maxRotationSpeed(0.005f)
+    , _mouseIsDown(false)
+    , _mouseIsMoving(false)
+    , _shiftHeld(false)
+{
+	FlyInit();
+}
+
+glm::mat4 Camera::GetRotationMatrix() const
 {
 	return glm::eulerAngleZXY(_rotation.z, _rotation.x, _rotation.y);
 }
 
 glm::mat4 Camera::GetViewMatrix() const
 {
-	return getRotationMatrix() * glm::translate(glm::mat4(1.0f), -_position);
+	return GetRotationMatrix() * glm::translate(glm::mat4(1.0f), -_position);
 }
 
 glm::mat4 Camera::GetViewProjectionMatrix() const
 {
 	return GetProjectionMatrix() * GetViewMatrix();
 }
 
+const std::optional<ecs::components::Transform> Camera::RaycastCamToLand()
```

Comment:
A note about returning const (not not const pointers). They don't prevent the caller from storing the result in a non-const copy. Therefore returning const is a useless qualifier.

On another note, it is good practice to set getters and functions that do not modify `this` as const. It gives the compiler hints for better optimization. 
https://www.geeksforgeeks.org/const-member-functions-c/

Reply:
realized it cant be const anyway as it modifies the state of `this`